src/plugin/file/zip/zipparserconfig.js

```diff
@@ -0,0 +1,58 @@
+goog.provide('plugin.file.zip.ZIPParserConfig');
+
+goog.require('os.parse.FileParserConfig');
+goog.require('os.ui.slick.column');
+goog.require('plugin.file.zip.ZIPParser');
+
+
+
+/**
+ * Configuration for a ZIP parser.
+ *
+ * @extends {os.parse.FileParserConfig}
+ * @constructor
+ */
+plugin.file.zip.ZIPParserConfig = function() {
+  plugin.file.zip.ZIPParserConfig.base(this, 'constructor');
+
+  /**
+   * @type {number}
+   */
+  this['status'] = -1;
+
+  /**
+   * @type {Array.<osx.import.FileWrapper>}
+   */
+  this['files'] = [];
+};
+goog.inherits(plugin.file.zip.ZIPParserConfig, os.parse.FileParserConfig);
+
+
+/**
+ * @param {Function} callback
+ */
+plugin.file.zip.ZIPParserConfig.prototype.update = function(callback) {
+  // re-initialize
+  if (this['files'].length > 0) this['files'] = [];
+  this['status'] = -1;
+
+  var parser = new plugin.file.zip.ZIPParser(this);
+
+  goog.events.listenOnce(parser, os.events.EventType.COMPLETE, goog.bind(function() {
```

Comment:
What happens if the parser errors instead of completing? I think this would be a memory leak in that scenario.

Reply:
Moved this code into zipparser.js > unzip() <line 149> and added ERROR throw on all the big errors from the parser with a catch here too.  Should cover everything but JS crashes.